chain/network/src/peer/peer_actor.rs

```diff
@@ -1009,15 +1011,18 @@ impl Handler<QueryPeerStats> for PeerActor {
     fn handle(&mut self, msg: QueryPeerStats, _: &mut Self::Context) -> Self::Result {
         #[cfg(feature = "delay_detector")]
         let _d = DelayDetector::new("query peer stats".into());
+
+        // TODO(#5218) Refactor this code to use `SystemTime`
+        let now = Instant::now();
+        let sent = self.tracker.sent_bytes.minute_stats(now);
+        let received = self.tracker.received_bytes.minute_stats(now);
+
         PeerStatsResult {
             chain_info: self.chain_info.clone(),
-            received_bytes_per_sec: self.tracker.received_bytes.bytes_per_min() / 60,
-            sent_bytes_per_sec: self.tracker.sent_bytes.bytes_per_min() / 60,
-            is_abusive: self.is_abusive(),
-            message_counts: (
-                self.tracker.sent_bytes.count_per_min(),
-                self.tracker.received_bytes.count_per_min(),
-            ),
+            received_bytes_per_sec: received.bytes_per_min / 60,
+            sent_bytes_per_sec: sent.bytes_per_min / 60,
+            is_abusive: self.is_abusive(now),
+            message_counts: (sent.count_per_min, received.count_per_min),
```

Comment:
Ultraminor nit, but I'd 

```
        let sent = self.tracker.sent_bytes.minute_stats(now);
        let received = self.tracker.received_bytes.minute_stats(now);
        let is_abusive = self.is_abusive(now);
```

to keep all statements where we look at time bunched up.

Reply:
Actually, now that I wrote it that way, I wonder if `is_abusive` needs to be a method on `self`? Like, the first thing it does is re-doing the computation we did on the previous line. That's the case where

```rust
let is_abusive = {
   // some comment for the block
   computation
};
```

pattern might we a win over an extracted function.

Reply:
Yes, inlining is a good idea. I updated the PR.